### PR TITLE
[CDAP-17034] Unit test for utilities function (Plugin JSON Creator)

### PR DESCRIPTION
**Pull request by jennac3**
Unit test for plugin JSON creator
- Convert state properties to JSON output correctly
- Convert JSON output to state properties correctly

https://builds.cask.co/browse/CDAP-UDUT728/latest

### CODE REVIEW

File: cdap-ui/app/cdap/components/AbstractWidget/AbstractWidgetFactory.tsx [hunk 1]
```diff
@@ -14,41 +14,41 @@
...
-import { objectQuery } from 'services/helpers';
 import CodeEditorWidget from 'components/AbstractWidget/CodeEditorWidget';
+import JsonEditorWidget from 'components/AbstractWidget/CodeEditorWidget/JsonEditorWidget';
 import CSVWidget from 'components/AbstractWidget/CSVWidget';
 import DatasetSelector from 'components/AbstractWidget/DatasetSelectorWidget';
 import DateRangeWidget from 'components/AbstractWidget/DateRangeWidget';
 import DateTimeWidget from 'components/AbstractWidget/DateTimeWidget';
-import FunctionDropdownAliasWidget from 'components/AbstractWidget/FunctionDropdownAliasWidget';
 import DLPCustomWidget from 'components/AbstractWidget/DLPCustomWidget';
+import MultiSelect from 'components/AbstractWidget/FormInputs/MultiSelect';
```
Comment (yukiej) [inline cdap-ui/app/cdap/components/AbstractWidget/AbstractWidgetFactory.tsx]: nit: the widget imports should be in alphabetical order

File: cdap-ui/app/cdap/components/AbstractWidget/__mocks__/AbstractWidgetFactory.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
```
Comment (yukiej) [inline cdap-ui/app/cdap/components/AbstractWidget/__mocks__/AbstractWidgetFactory.tsx]: nit: update copyright

File: cdap-ui/app/cdap/components/PluginJSONCreator/__tests__/plugin-json-creator-utilities.test.ts [hunk 3]
```diff
@@ -0,0 +1,378 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
```
Comment (yukiej) [inline cdap-ui/app/cdap/components/PluginJSONCreator/__tests__/plugin-json-creator-utilities.test.ts]: Nit: update copyright date

File: cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/ConfigurationGroupPage/GroupPanel/WidgetCollection/WidgetPanel/WidgetInfoInput/index.tsx [hunk 4]
```diff
@@ -47,6 +47,7 @@ const WidgetInfoInputView: React.FC<IWidgetInfoInputProps> = ({ classes, widgetI
 
   function onNameChange() {
     return (name) => {
+      console.log('name', name);
```
Comment (yukiej) [inline cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/ConfigurationGroupPage/GroupPanel/WidgetCollection/WidgetPanel/WidgetInfoInput/index.tsx]: does this need to be here?

File: cdap-ui/app/cdap/components/PluginJSONCreator/utilities/index.ts [hunk 5]
```diff
@@ -1,9 +1,16 @@
-import { SchemaType, SPEC_VERSION } from 'components/PluginJSONCreator/constants';
+// import { SchemaType, SPEC_VERSION } from 'components/PluginJSONCreator/constants';
```
Comment (yukiej) [inline cdap-ui/app/cdap/components/PluginJSONCreator/utilities/index.ts]: Can these comments be deleted?

**Review by yukiej — APPROVED** [review]
A few minor comments,  but otherwise lgtm!

File: cdap-ui/app/cdap/components/AbstractWidget/__mocks__/AbstractWidgetFactory.tsx (hunk 2 above)
Comment (jennac3) [reply cdap-ui/app/cdap/components/AbstractWidget/__mocks__/AbstractWidgetFactory.tsx]: Resolved

File: cdap-ui/app/cdap/components/PluginJSONCreator/__tests__/plugin-json-creator-utilities.test.ts (hunk 3 above)
Comment (jennac3) [reply cdap-ui/app/cdap/components/PluginJSONCreator/__tests__/plugin-json-creator-utilities.test.ts]: Resolved

File: cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/ConfigurationGroupPage/GroupPanel/WidgetCollection/WidgetPanel/WidgetInfoInput/index.tsx (hunk 4 above)
Comment (jennac3) [reply cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/ConfigurationGroupPage/GroupPanel/WidgetCollection/WidgetPanel/WidgetInfoInput/index.tsx]: Resolved

File: cdap-ui/app/cdap/components/PluginJSONCreator/utilities/index.ts (hunk 5 above)
Comment (jennac3) [reply cdap-ui/app/cdap/components/PluginJSONCreator/utilities/index.ts]: Resolved

**Pull request merged**